src/lib/FreeC/IR/Similar.hs

```diff
@@ -337,7 +343,9 @@ similarExpr (IR.Case _ e as _) (IR.Case _ f bs _) =
 similarExpr (IR.If _ e1 e2 e3 _) (IR.If _ f1 f2 f3 _) =
   similar' e1 f1 .&&. similar' e2 f2 .&&. similar' e3 f3
 similarExpr (IR.Let _ bs e _) (IR.Let _ cs f _) =
-  similar' e f .&&. similar' bs cs
+  let ns = map (IR.varPatQName . IR.bindVarPat) bs
+      ms = map (IR.varPatQName . IR.bindVarPat) cs
+  in similar' e f . extendRenaming IR.ValueScope ns ms .&&. similar' bs cs
```

Comment:
Since the `Similar` instance for `IR.Bind` extends the renaming by the variable bound by the binding only, this corresponds to

```
Γ ∪ { xᵢ ↦ yᵢ } ⊢ bᵢ ≈ dᵢ
```

instead of

```
Γ' ⊢ bᵢ ≈ dᵢ
```

If you remove `extendRenaming` from the `Similar` instance of `IR.Bind`, the following should work.

```suggestion
  in  (similar' e f .&&. similar' bs cs) . extendRenaming IR.ValueScope ns ms
```